tests/src/python/test_qgsserver_wms_dxf.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+"""QGIS Unit tests for QgsServer WMS GetPrint.
+
+From build dir, run: ctest -R PyQgsServerWMSGetMapDxf -V
+
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+"""
+__author__ = 'Tudor Bărăscu'
+__date__ = '27/01/2021'
+__copyright__ = 'Copyright 2020, The QGIS Project'
+
+import os
+
+import urllib.parse
+
+from qgis.testing import unittest
+
+from test_qgsserver import QgsServerTestBase
+
+from qgis.core import (QgsVectorLayer)
+
+# Needed on Qt 5 so that the serialization of XML is consistent among all executions
+os.environ['QT_HASH_SEED'] = '1'
+
+
+class PyQgsServerWMSGetMapDxf(QgsServerTestBase):
+
+    def test_dxf_export_works_with_reverse_axis_epsg(self):
+        qs = "?" + "&".join(["%s=%s" % i for i in list({
+            "MAP": urllib.parse.quote(os.path.join(self.testdata_path,
+                                                   'test_dxf_export.qgs')),
+            "SERVICE": "WMS",
+            "VERSION": "1.3.0",
+            "REQUEST": "GetMap",
+            "BBOX": "399980,449980,400050,450100",
+            "CRS": "EPSG:3844",
+            "LAYERS": "test_dxf_export",
+            "STYLES": ",",
+            "FORMAT": "application/dxf",
+            "SCALE": "500",
+            "FILE_NAME": "test_dxf_export.dxf"
+        }.items())])
+
+        r, h = self._result(self._execute_request(qs))
+
+        dxf = '/tmp/test_dxf_export.dxf'
```

Comment:
Please use QTemporaryDir or python's tempfile for temporary test output

Reply:
Done